src/benchmarking/directory_parser.py

```diff
@@ -0,0 +1,207 @@
+    BRACKEN_NO_ASSEMBLY = 'bracken/nonassembled_abundances.bracken'
+
+
+def parse_directory_trees(truth_path: str, result_path: str, output_prefix: str) -> None:
+    '''
+    DESCRIPTION:
+        Get directory structure metadata from file paths and write to CSV.
+        Handles truth and result directory trees
+    
+    INPUT:
+        1. Path to truth directory tree
+        2. Path to results directory tree (i.e. the results of the tools that were tested) 
+        3. Path to desired place to store CSV file
+
+    OUTPUT:
+        CSV file containing: <trial #>,<tool>,<test>,<subtest>,<path to final result file>
+        e.g. 3,EnrichSeq,num_genomes,10_genomes_500000_reads,/projects/benchmarking/results/test_3/EnrichSeq/num_genomes/10_genomes_500000_reads/enrichseq/output_files/taxid_abundances.csv
+    '''
+    
+    # create CSV file to append to
+    output_csv_path = Path(settings.OUTPUT_DIRECTORY) / Path(output_prefix + '.csv')
+
+    # if os.path.exists(output_csv_path):
+    #     os.remove(output_csv_path)
+
+    with open(output_csv_path, 'w') as csvfile:
+        writer = csv.writer(csvfile) 
+        writer.writerow(['Trial_Num', 'Experiment', 'Condition', 'Tool', 'File_Path'])
+
+    # parse truth
+    truth_rows = parse_truth_filepaths(truth_path)
+    file = write_to_csv(truth_rows, output_csv_path)
+
+    # parse results
+    result_rows = parse_results_filepaths(result_path)
+    file = write_to_csv(result_rows, output_csv_path)
+    
+    print(f'Metadata saved to:{file}')
+
+
+def parse_truth_filepaths(dir_path: str) -> list:
+    '''
+    DESCRIPTION:
+        Get metadata from truth directory paths and store in list
+        
+    INPUT:
+        Path to all truth files (called "tests" in our case)
+    
+    OUTPUT:
+        A list of lists. Each row contains the metadata for each "truth" entry
+    '''
+
+    rows = []
+
+    for trial in os.listdir(dir_path):
+        if trial.startswith('.'):
+            continue
+        try:
+            trial_num = int(''.join(filter(str.isdigit, trial)))
+        except ValueError as e:
+            trial_num = 0
+            print(f"Directory {trial} does not have an integer, causing error: {e}", file=sys.stderr)
+        finally:
+            trial_path = Path(dir_path) / trial
+            if os.path.isdir(trial_path):
+                for test in os.listdir(trial_path):
+                    test_path = Path(trial_path) / test
+                    if os.path.isdir(test_path) and 'config' not in test:
+                        for subtest in os.listdir(test_path):
+                            subtest_path = Path(test_path) / subtest
+                            if os.path.isfile(subtest_path):
+                                if subtest.endswith('.fa'):
+                                    info = [trial_num, test, subtest.strip('.fa'), 'TRUTH', os.path.abspath(subtest_path)] 
+                                    rows.append(info)
```

Comment:
This could be made into 2 lines using a recursive scan (think BFS on the tree; or inOrder traversal).